agent.go

```diff
@@ -856,12 +895,36 @@ func (p *pod) runContainerProcess(cid, pid string, terminal bool, started chan e
 
 	fieldLogger := agentLog.WithField("container-pid", pid)
 
+	// This lock is very important to avoid any race with reaper.reap().
+	// Indeed, if we don't lock this here, we could potentially get the
+	// SIGCHLD signal before the channel has been created, meaning we will
```

Comment:
Can you mention exit code channel to be more clear.

Reply:
Ok.